mobile/src/main/java/ch/epfl/sdp/mobile/ui/setting/SettingScreen.kt

```diff
@@ -75,7 +77,14 @@ fun SettingHeader(state: SettingScreenState, modifier: Modifier = Modifier) {
   ) {
     SettingPicture(state)
     Column(horizontalAlignment = Alignment.CenterHorizontally) {
-      Text(state.name, style = MaterialTheme.typography.h5)
+      Row(horizontalArrangement = Arrangement.Center) {
+        Text(state.name, style = MaterialTheme.typography.h5)
+        IconButton(
+            onClick = state::openEditProfileName,
+            modifier = Modifier.size(35.dp).testTag("editProfileName")) {

```

Comment:
Material Design is aligned on an 8 Dp grid. Maybe try with `24.dp` instead ? It's the standard icon size.

Also, you should probably apply the size modifier on the icon, not the wrapper `IconButton`.